src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -119,5 +142,61 @@ private CodeAction TryCreateCodeAction(Document document, AddImportFixData fixDa
 
             throw ExceptionUtilities.Unreachable;
         }
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return new AddImportCodeFixAllProvider();
+        }
+
+        internal async Task<Document> FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var getFixesTasks = diagnostics
+                .GroupBy(d => d.Location.SourceSpan)
+                .Select(g => GetFixesForDiagnosticsAsync(document, g.Key, g.AsImmutable(), true, false, cancellationToken));
+
+            var usableFixes = (await Task.WhenAll(getFixesTasks).ConfigureAwait(false))
+                .SelectMany(diagnosticFixes => diagnosticFixes)
+                // Diagnostics where there are more than one fix signify an ambiguity that we should leave for the user to sort out.
+                .Where(diagnosticFixes => diagnosticFixes.Fixes.Length == 1)
+                .Select(diagnosticFixes => diagnosticFixes.Fixes[0])
+                .Distinct(this)
+                .Reverse();
+
+            foreach (var fix in usableFixes)
+            {
+                var codeAction = TryCreateCodeAction(document, fix);
+                if (codeAction == null)
+                {
+                    continue;
+                }
+
+                var newSolution = await codeAction.GetChangedSolutionAsync(new ProgressTracker(), cancellationToken).ConfigureAwait(false);
```

Comment:
this feels.. a bit wierd.. what doing it one by one?

Reply:
Since all of these code actions work against the same document (and sometimes project), is there a way to perform them in parallel and merge the results afterwards?

Reply:
@JoeRobich .  Yes: https://github.com/dotnet/roslyn/pull/28029